<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>MountDialog</class>
 <widget class="QDialog" name="MountDialog">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>500</width>
    <height>400</height>
   </rect>
  </property>
  <property name="windowTitle">
   <string>Mount Encrypted Drive</string>
  </property>
  <layout class="QVBoxLayout" name="verticalLayout">
   <item>
    <widget class="QLabel" name="promptLabel">
     <property name="text">
      <string>Enter password to mount the encrypted drive:</string>
     </property>
    </widget>
   </item>
   <item>
    <widget class="QLabel" name="helpLabel">
     <property name="text">
      <string/>
     </property>
     <property name="wordWrap">
      <bool>true</bool>
     </property>
    </widget>
   </item>
   <item>
    <layout class="QFormLayout" name="formLayout">
     <item row="0" column="0">
      <widget class="QLabel" name="driveLabel">
       <property name="text">
        <string>Drive File:</string>
       </property>
      </widget>
     </item>
     <item row="0" column="1">
      <layout class="QHBoxLayout" name="driveLayout">
       <item>
        <widget class="QLineEdit" name="drivePathEdit">
         <property name="placeholderText">
          <string>Select encrypted drive file</string>
         </property>
        </widget>
       </item>
       <item>
        <widget class="QPushButton" name="browseDriveButton">
         <property name="text">
          <string>Browse...</string>
         </property>
        </widget>
       </item>
      </layout>
     </item>
     <item row="1" column="0">
      <widget class="QLabel" name="mountLabel">
       <property name="text">
        <string>Mount Point:</string>
       </property>
      </widget>
     </item>
     <item row="1" column="1">
      <layout class="QHBoxLayout" name="mountLayout">
       <item>
        <widget class="QLineEdit" name="mountPointEdit">
         <property name="placeholderText">
          <string>Select mount location</string>
         </property>
        </widget>
       </item>
       <item>
        <widget class="QPushButton" name="browseMountButton">
         <property name="text">
          <string>Browse...</string>
         </property>
        </widget>
       </item>
      </layout>
     </item>
     <item row="2" column="0">
      <widget class="QLabel" name="passwordLabel">
       <property name="text">
        <string>Password:</string>
       </property>
      </widget>
     </item>
     <item row="2" column="1">
      <widget class="QLineEdit" name="passwordEdit">
       <property name="echoMode">
        <enum>QLineEdit::Password</enum>
       </property>
      </widget>
     </item>
    </layout>
   </item>
   <item>
    <widget class="QCheckBox" name="savePasswordCheckBox">
     <property name="text">
      <string>Save password for this drive</string>
     </property>
    </widget>
   </item>
   <item>
    <widget class="QCheckBox" name="keyFileCheckBox">
     <property name="text">
      <string>Use key file</string>
     </property>
    </widget>
   </item>
   <item>
    <widget class="QCheckBox" name="readOnlyCheckBox">
     <property name="text">
      <string>Mount as read-only</string>
     </property>
    </widget>
   </item>
   <item>
    <widget class="QCheckBox" name="advancedCheckBox">
     <property name="text">
      <string>Show Advanced Options</string>
     </property>
    </widget>
   </item>
   <item>
    <widget class="QGroupBox" name="advancedGroupBox">
     <property name="title">
      <string>Advanced Options</string>
     </property>
     <property name="visible">
      <bool>false</bool>
     </property>
     <layout class="QVBoxLayout" name="advancedLayout">
      <item>
       <widget class="QCheckBox" name="cacheCheckBox">
        <property name="text">
         <string>Enable in-memory cache</string>
        </property>
       </widget>
      </item>
      <item>
       <layout class="QHBoxLayout" name="cacheSizeLayout">
        <item>
         <widget class="QLabel" name="cacheSizeLabel">
          <property name="text">
           <string>Cache Size (MB):</string>
          </property>
         </widget>
        </item>
        <item>
         <widget class="QSpinBox" name="cacheSizeSpinBox">
          <property name="enabled">
           <bool>false</bool>
          </property>
          <property name="minimum">
           <number>32</number>
          </property>
          <property name="maximum">
           <number>4096</number>
          </property>
          <property name="value">
           <number>256</number>
          </property>
         </widget>
        </item>
        <item>
         <spacer name="horizontalSpacer">
          <property name="orientation">
           <enum>Qt::Horizontal</enum>
          </property>
          <property name="sizeHint" stdset="0">
           <size>
            <width>40</width>
            <height>20</height>
           </size>
          </property>
         </spacer>
        </item>
       </layout>
      </item>
     </layout>
    </widget>
   </item>
   <item>
    <spacer name="verticalSpacer">
     <property name="orientation">
      <enum>Qt::Vertical</enum>
     </property>
     <property name="sizeHint" stdset="0">
      <size>
       <width>20</width>
       <height>40</height>
      </size>
     </property>
    </spacer>
   </item>
   <item>
    <widget class="QDialogButtonBox" name="buttonBox">
     <property name="orientation">
      <enum>Qt::Horizontal</enum>
     </property>
     <property name="standardButtons">
      <set>QDialogButtonBox::Cancel|QDialogButtonBox::Ok</set>
     </property>
    </widget>
   </item>
  </layout>
 </widget>
 <resources/>
 <connections>
  <connection>
   <sender>buttonBox</sender>
   <signal>accepted()</signal>
   <receiver>MountDialog</receiver>
   <slot>accept()</slot>
  </connection>
  <connection>
   <sender>buttonBox</sender>
   <signal>rejected()</signal>
   <receiver>MountDialog</receiver>
   <slot>reject()</slot>
  </connection>
  <connection>
   <sender>cacheCheckBox</sender>
   <signal>toggled(bool)</signal>
   <receiver>cacheSizeSpinBox</receiver>
   <slot>setEnabled(bool)</slot>
  </connection>
  <connection>
   <sender>advancedCheckBox</sender>
   <signal>toggled(bool)</signal>
   <receiver>advancedGroupBox</receiver>
   <slot>setVisible(bool)</slot>
  </connection>
 </connections>
</ui>
